express-notes-api/index.js

```diff
@@ -0,0 +1,91 @@
+const data = require('./data.json');
+const express = require('express');
+const fs = require('fs');
+const app = express();
+
+app.listen(3000, () => {
+});
+
+app.get('/api/notes', (req, res) => {
+  const notes = [];
+  for (const key in data.notes) {
+    notes.push(data.notes[key]);
+  }
+  res.json(notes);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  let note = {};
+  const id = req.params.id;
+  if (req.params.id < 1) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (req.params.id) {
+    note = data.notes[req.params.id];
+    if (note === undefined) {
+      res.status(404).json({ error: 'cannot find note with id ' + id });
+    }
+  }
+  res.json(note);
```

Comment:
Make sure to use `else` or `return` appropriately to prevent your server from attempting to respond more than once to one request. Double-check your other routes too.